### PR TITLE
Worker versioning API Changes

### PR DESCRIPTION
**Pull request by Sushisource**
**What changed?**
We've been working in a branch for some time on API changes for versioning. Everything here has already been reviewed independently.

**Why?**
Versioning feature

**Breaking changes**
Only to existing-but-unimplemented RPCs

### CODE REVIEW

File: temporal/api/workflow/v1/message.proto [hunk 1]
```diff
@@ -109,8 +109,12 @@ message ResetPoints {
 }
 
 message ResetPointInfo {
+    // A worker binary version identifier, will be deprecated and superceeded by a newer concept of build_id (see
+    // below).
```
Comment (bergundy) [inline temporal/api/workflow/v1/message.proto:113]:
```suggestion
    // A worker binary version identifier, will be deprecated and superceeded by a newer concept of build_id.
```

Comment (Sushisource) [reply temporal/api/workflow/v1/message.proto:113]: Heh, you wrote this line! (And spelled superseded wrong) :stuck_out_tongue:

File: temporal/api/errordetails/v1/message.proto [hunk 2]
```diff
@@ -106,3 +106,8 @@ message SystemWorkflowFailure {
 
 message WorkflowNotReadyFailure {
 }
+
+message NewerBuildExistsFailure {
+    // Build ID of the newer compatible build that will receive tasks.
+    string latest_build_id = 1;
```
Comment (dnr) [inline temporal/api/errordetails/v1/message.proto:112]: should we rename this to `default` or `newer`?

File: temporal/api/history/v1/message.proto [hunk 3]
```diff
@@ -102,6 +102,12 @@ message WorkflowExecutionStartedEventAttributes {
     int64 parent_initiated_event_version = 26;
     // This field is new in 1.21.
     string workflow_id = 28;
+    // When using build-id versioning, we tie the version of a child workflow or
+    // continued-as-new workflow on the same task queue to the parent/previous workflow by
+    // default (unless requested with use_latest_build_id). To make that version
```
Comment (dnr) [inline temporal/api/history/v1/message.proto:107]: should update this

File: temporal/api/history/v1/message.proto [hunk 4]
```diff
@@ -297,6 +316,9 @@ message ActivityTaskCompletedEventAttributes {
     int64 started_event_id = 3;
     // id of the worker that completed this task
     string identity = 4;
+    // Version info of the worker who processed this activity task, or missing if worker is not
+    // using versioning.
```
[the remark this answers is not in the document]
Comment (Sushisource) [reply temporal/api/history/v1/message.proto:320]: Yeah, it isn't quite true anymore. At least how I implemented it in go it is always populated unless the activity was completed async by a client that doesn't have a build id

File: temporal/api/errordetails/v1/message.proto (hunk 2 above)
Comment (Sushisource) [reply temporal/api/errordetails/v1/message.proto]: Did this

**Review by dnr — APPROVED** [review]

**Pull request merged**